### PR TITLE
Added initial code for VP9 support.

### PR DESCRIPTION
**Pull request by jeffqjiangNew**
- Added stream syntax parsing.
- Added VA-API decode call.
- Intra decode started to work.

### CODE REVIEW

File: src/parser/vp9_parser.cpp [hunk 1]
```diff
@@ -0,0 +1,873 @@
...
+    p_pic_param->pic_fields.bits.alt_ref_frame_sign_bias = p_uncomp_header->ref_frame_sign_bias[kVp9AltRefFrame];
+    p_pic_param->pic_fields.bits.lossless_flag = p_uncomp_header->quantization_params.lossless;
+
+    p_pic_param->filter_level = p_uncomp_header->loop_filter_params.loop_filter_level;
+    p_pic_param->sharpness_level = p_uncomp_header->loop_filter_params.loop_filter_sharpness;
+    p_pic_param->log2_tile_rows = p_uncomp_header->tile_info.tile_rows_log2;
+    p_pic_param->log2_tile_columns = p_uncomp_header->tile_info.tile_cols_log2;
+    p_pic_param->frame_header_length_in_bytes = uncomp_header_size_;
+    p_pic_param->first_partition_size = p_uncomp_header->header_size_in_bytes;
+    for( int i = 0; i < 7; i++) {
```
Comment (AryanSalmanpour) [inline src/parser/vp9_parser.cpp]: minor formatting: `for( int` ->` for (int`

File: src/parser/vp9_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,873 @@
...
+    curr_pic_.dec_buf_idx = dec_buf_index;
+    decode_buffer_pool_[dec_buf_index].use_status |= kFrameUsedForDecode;
+    // Todo decode_buffer_pool_[dec_buf_index].pic_order_cnt = curr_pic_.order_hint;
+    decode_buffer_pool_[dec_buf_index].pts = curr_pts_;
+    return PARSER_OK;
+}
+
+ParserResult Vp9VideoParser::FindFreeInDpbAndMark() {
+    int i;
+    /*Todo for (i = 0; i < VP9_NUM_REF_FRAMES; i++ ) {
```
Comment (AryanSalmanpour) [inline src/parser/vp9_parser.cpp]: are you planning to finish this function in the current PR or next?

File: src/parser/vp9_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,873 @@
...
+        }
+    } else {
+        p_uncomp_header->segmentation_params.segmentation_update_map = 0;
+        p_uncomp_header->segmentation_params.segmentation_temporal_update = 0;
+    }
+}
+
+uint8_t Vp9VideoParser::ReadProb(const uint8_t *p_stream, size_t &offset) {
+    uint8_t prob_coded;
+    uint8_t prob;
```
Comment (AryanSalmanpour) [inline src/parser/vp9_parser.cpp]: You can initialize the `prob` variable to 255 here and remove the else condition below.

**Review by AryanSalmanpour — COMMENTED** [review]
I added a few comments. Please take a look. Thanks.

File: src/parser/vp9_parser.cpp (hunk 1 above)
Comment (jeffqjiangNew) [reply src/parser/vp9_parser.cpp]: Done.

File: src/parser/vp9_parser.cpp (hunk 2 above)
Comment (jeffqjiangNew) [reply src/parser/vp9_parser.cpp]: Moved the commented code and will add in the next PR.

File: src/parser/vp9_parser.cpp (hunk 3 above)
Comment (jeffqjiangNew) [reply src/parser/vp9_parser.cpp]: Yes that is cleaner. Done.

**Review by AryanSalmanpour — APPROVED** [review]

**Pull request merged**